packages/web/src/hooks/usePrompt.ts

```diff
@@ -56,12 +56,13 @@ const applyNavigatorBlockers = (navigator: Navigator): NavigatorWithBlockers =>
 export const usePrompt = (message: string, when: BlockerCondition = () => false) => {
   const {navigator} = useContext(NavigationContext);
   const enhancedNavigator = useMemo(() => applyNavigatorBlockers(navigator), [navigator]);
-  const rule = useLastCallback(when);
-
   const unsubscribe = useRef(() => {});
-  const prevMessage = usePrevious(message);
+  const rule = useLastCallback(when);
 
   useEffect(() => {
+    unsubscribe.current();
+    unsubscribe.current = enhancedNavigator.$block(message, rule);
```

Comment:
This way, it won't unsubscribe after destroying component with `usePrompt`. When it will end up blocking, it won't every unblock after destroying.